torch/nn/utils/prune.py

```diff
@@ -0,0 +1,1269 @@
+        # the original tensor to a new parameter called name + '_orig' and
+        # and deleting the original parameter
+        if not isinstance(method, PruningContainer):
+            # copy `module[name]` to `module[name + '_orig']`
+            module.register_parameter(name + "_orig", orig)
+            # temporarily delete `module[name]`
+            del module._parameters[name]
+            default_mask = torch.ones_like(orig)  # temp
+        # If this is not the first time pruning is applied, all of the above
+        # has been done before in a previos pruning iteration, so we're good
+        # to go
+        else:
+            default_mask = getattr(module, name + "_mask").detach().clone()
+
+        # Use try/except because if anything goes wrong with the mask 
+        # computation etc., you'd want to roll back.
+        try:
+            # get the final mask, computed according to the specific method
+            mask = method.compute_mask(orig, default_mask=default_mask)
+            # reparametrize by saving mask to `module[name + '_mask']`...
+            module.register_buffer(name + "_mask", mask)
+            # ... and the new pruned tensor to `module[name]`
+            setattr(module, name, method.apply_mask(module))
+            # associate the pruning method to the module via a hook to
+            # compute the function before every forward() (compile by run)
+            module.register_forward_pre_hook(method)
+
+        except Exception as e:
+            print(type(method))
+            if not isinstance(method, PruningContainer):
+                orig = getattr(module, name + "_orig")
+                module.register_parameter(name, orig)
+                del module._parameters[name + "_orig"]
+            raise e
+
+        return method
+
+    def remove(self, module):
+        r"""Removes the pruning reparameterization from a module. The pruned
+        parameter named `name` remains permanently pruned, and the parameter
+        named `name+'_orig'` is removed from the parameter list. Similarly,
+        the buffer named `name+'_mask'` is removed from the buffers.
+
+        Note: 
+            Pruning itself is NOT undone or reversed!
+        """
+        # before removing pruning from a tensor, it has to have been applied
+        assert (
+            self._tensor_name is not None
+        ), "Module {} has to be pruned\
+            before pruning can be removed".format(
+            module
+        )  # this gets set in apply()
+
+        # to update module[name] to latest trained weights
+        weight = self.apply_mask(module)  # masked weights
+
+        # delete and reset
+        delattr(module, self._tensor_name)
+        orig = module._parameters[self._tensor_name + "_orig"]
+        orig.data = weight.data
+        del module._parameters[self._tensor_name + "_orig"]
+        del module._buffers[self._tensor_name + "_mask"]
+        module.register_parameter(self._tensor_name, orig)
```

Comment:
It will change the ordering, yes. `_parameters` is an `OrderedDict`, so yes, it will know that I removed something and then added something new, and it will propagate that to the way `.parameters()` yields things.
Whether one _should_ rely on `.parameters()` being ordered is probably not something I can judge :) But, in case you are cool with supporting that usage, I can always brute force it:
```
d = OrderedDict(['a':1, 'b':2, 'c':3])
d = OrderedDict([('b_new', -2) if k == 'b' else (k, v) for k,v in d.items()])
print(d)
>>> OrderedDict([('a', 1), ('b_new', -2), ('c', 3)])
```
LMK what you want me to do about this.

Reply:
@gchanan do you know if anyone relies on ordering?